pkg/controller/gather_commands.go

```diff
@@ -424,65 +422,85 @@ func wasDataProcessed(ctx context.Context,
-			return false, nil
-		}
+		return processSuccessfulResp(resp.Body, &statusRetryCounter, &dataProcessed, delay)
+	})
+	if err != nil {
+		return false, err
+	}
 
-		data, err := io.ReadAll(resp.Body)
-		if err != nil {
-			return false, err
-		}
+	return dataProcessed, nil
+}
 
-		var processingResp dataStatus
-		err = json.Unmarshal(data, &processingResp)
-		if err != nil {
-			return false, err
-		}
+// networkRetry is used to retry when there is a network issue that caused failure
+func networkRetry(networkRetryCounter *int, err error, delay time.Duration) error {
+	if *networkRetryCounter >= numberOfStatusQueryRetries {
+		return fmt.Errorf("failed to check processing status after %d retries: %w", *networkRetryCounter, err)
+	}
+	klog.Infof("Network error when checking processing status: %v, retry %d/%d in %s",
+		err, *networkRetryCounter+1, numberOfStatusQueryRetries, delay)
+	*networkRetryCounter++
+	return nil
+}
 
-		dataProcessed = (processingResp.Status == "processed")
-
-		// If data is not processed yet, retry 3 times before failing
-		if !dataProcessed {
-			if statusRetryCounter >= numberOfStatusQueryRetries {
-				klog.Infof("Data status is %q after %d retries, stopping poll", processingResp.Status, statusRetryCounter)
-				return false, fmt.Errorf("data processing status is %q after %d retries, stopping poll", processingResp.Status, statusRetryCounter)
-			}
-			klog.Infof("Data status is %q, retry %d/%d in %s",
-				processingResp.Status, statusRetryCounter+1, numberOfStatusQueryRetries, delay)
-			statusRetryCounter++
-			return false, nil
-		}
+// requestRetry is used to retry an http request when the response != 200
+func requestRetry(requestRetryCounter *int, respStatusCode int, delay time.Duration) error {
+	if *requestRetryCounter >= numberOfStatusQueryRetries {
+		return fmt.Errorf("HTTP status message: %s", http.StatusText(respStatusCode))
+	}
+	klog.Infof("Received HTTP status code %d, retry %d/%d in %s",
+		respStatusCode, *requestRetryCounter+1, numberOfStatusQueryRetries, delay)
+	*requestRetryCounter++
+	return nil
+}
 
-		return true, nil
-	})
+// processSuccessfulResp is used to process response body and if data is not processed, it retries 3 times
+// Returns true if the data was successfully processed, false otherwise
+func processSuccessfulResp(respBody io.ReadCloser, statusRetryCounter *int, dataProcessed *bool, delay time.Duration) (bool, error) {
+	if respBody == nil || respBody == http.NoBody {
+		return false, nil
+	}
+
+	data, err := io.ReadAll(respBody)
 	if err != nil {
 		return false, err
 	}
 
-	return dataProcessed, nil
+	var processingResp dataStatus
+	err = json.Unmarshal(data, &processingResp)
+	if err != nil {
+		return false, err
+	}
+
+	*dataProcessed = (processingResp.Status == "processed")
+
+	// If data is not processed yet, retry 3 times before failing
+	if !*dataProcessed {
+		return statusRetry(statusRetryCounter, processingResp.Status, delay)
+	}
```

Comment:
If we drop dataProcessed, we could do this:
```suggestion
	if processingResp.Status != "processed" {
		return statusRetry(statusRetryCounter, processingResp.Status, delay)
	}
```